functions/_nvm_use_on_pwd_change.fish

```diff
@@ -0,0 +1,9 @@
+function _nvm_use_on_pwd_change \
+    --on-variable nvm_use_on_pwd_change \
+    --on-variable PWD \
+    --description 'Use Node.js version specified by project automatically'
+
+    if test "$nvm_use_on_pwd_change" = true
+        nvm use 2>/tmp/_nvm_use_on_pwd_change.log
+    end
+end && _nvm_use_on_pwd_change
```

Comment:
We'll end up running the function whenever the file is sourced.

Reply:
ok the `--on-variable` option is enough, my bad